File: 1010-Pairs of Songs With Total Durations Divisible by 60.py
from collections import defaultdict
class Solution(object):
    def numPairsDivisibleBy60(self, time):
        """
        :type time: List[int]
        :rtype: int
        """
        hmap = defaultdict(int)
        count = 0
        
        for t in time:
            pair = 60 - t%60
            if pair == 60:
                pair = 0
            count += hmap[pair]
            hmap[t%60] += 1
        
        return count

File: JavaScript/1010-Pairs of Songs With Total Durations Divisible by 60.js
/**
 * @param {number[]} time
 * @return {number}
 */
var numPairsDivisibleBy60 = function(time) {
    
    let count = 0;
    let hmap = new Map();

    for (let t of time){
        let pair = 60 - t%60;
        if (pair === 60){
            pair = 0;
        }
        count += hmap.get(pair) || 0;
        hmap.set(t%60, (hmap.get(t%60)||0)+1);
    }
    return count;
};